lesson15/js/index.js

```diff
@@ -0,0 +1,102 @@
+const modalBtn = document.getElementById("js-modal-btn");
+const modal = document.getElementById("js-modal");
+const requestBtn = document.getElementById("js-request-btn");
+const ul = document.getElementById("js-ul");
+const modalCloseIcon = document.getElementById('js-modal-close-icon');
+
+function addLoading() {
+  const wrapper = document.getElementById("js-wrapper");
+  const loading = document.createElement("img");
+  loading.src = "./img/loading-circle.gif";
+  loading.id = "loading";
+  wrapper.appendChild(loading);
+}
+
+function removeLoading() {
+  const loading = document.getElementById("loading");
+  loading.remove();
+}
+
+async function fetchErrorHandling(response) {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+async function getJsonOrError() {
+  const response = await fetch(
+    "https://jsondata.okiba.me/v1/json/fR9k5211007080400"
+  );
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+function createLists({ data }) {
+  const frag = document.createDocumentFragment();
+  data.forEach(value => {
+    const { a, img, alt, text } = value;
+    const li = document.createElement("li");
+    const anchor = document.createElement("a");
+    const image = document.createElement("img");
+
+    anchor.href = `/${a}.html`;
+    image.src = img;
+    image.alt = alt;
+
+    frag.appendChild(li).appendChild(anchor).appendChild(image);
+    anchor.insertAdjacentHTML("beforeend", text);
+  });
+  ul.appendChild(frag);
+}
+
+async function tryGetData() {
+  addLoading();
+  try {
+    return await getJsonOrError();
+  } catch (error) {
+    ul.replaceWith(error);
+  } finally {
+    removeLoading();
+  }
+}
+
+async function tryCreate() {
+  try {
+    const responseData = await tryGetData();
+    if (responseData.data) {
+      createLists(responseData);
+    } else {
+      throw new Error("適切なデータが見つかりませんでした");
+    }
+  } catch (error) {
+    ul.replaceWith(error);
+  }
+}
+
+modalBtn.addEventListener("click", (e) => {
+  e.target.style.display = "none";
+  modal.classList.add("visible");
+});
+
+requestBtn.addEventListener("click", (e) => {
+  const inputName = document.getElementById("name").value;
+  const inputNumber = document.getElementById("number").value;
+  e.preventDefault();
+
+  const trimInputName = inputName.trim();
+
+  if (trimInputName.length === 0 || inputNumber === "") {
```

Comment:
nits
I think the empty string is the problem, not the length.

```suggestion
  if (trimmedInputName === "" || inputNumber === "") {
```

Reply:
Thanks for the review!
I agree with your opinion.
I felt it would be easier to understand if the same conditions were used as inputNumber.
Also, there was no reason to stick with "length".

faeb57b